### PR TITLE
Baseline rewriting

### PR DESCRIPTION
**Pull request by pmderodat**

### CODE REVIEW

File: src/e3/testsuite/report/rewriting.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+        :param test_name: Name of the test for which we want the baseline
+            filename.
+        """
+        ...
+
+    def postprocess_baseline(self, baseline: bytes) -> bytes:
+        """Refine a baseline to rewrite.
+
+        By default, this returns the argument unchnaged. Subclasses can
```
Comment (kanigsson) [inline src/e3/testsuite/report/rewriting.py:64]: unchanged

File: src/e3/testsuite/report/rewriting.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                # In case the output is a string, we need its encoding to write
+                # it down to a file.
+                assert result.env is not None
+                encoding = result.env.get("encoding", self.default_encoding)
+
+                if result.out is None:
+                    self.handle_test_error(
+                        summary,
+                        test_name,
+                        "no output associtaed to test result",
```
Comment (kanigsson) [inline src/e3/testsuite/report/rewriting.py:125]: associated

**Review by kanigsson — APPROVED** [review]
Just some small typos. I'm not exactly an expert in e3-testsuite, but I'm eager to try it out :-)

**Review by yakobowski — COMMENTED** [review]
Very nice! Thank you.

**Comment by pmderodat** [comment]
Thank you both for the review! All 3 typos should fixed.

**Review by kanigsson — APPROVED** [review]

File: src/e3/testsuite/report/rewriting.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            new_baseline
+            if isinstance(new_baseline, bytes)
+            else new_baseline.encode(encoding)
+        )
+        if baseline_bytes:
+            if not baseline_exists:
+                self.print_info(
+                    f"no baseline file for {test_name}, creating it"
+                )
+            summary.new_baselines.add(test_name)
```
Comment (yakobowski) [inline src/e3/testsuite/report/rewriting.py:214]: This line should be in the `if` above @pmderodat. Sorry for having missed that.

Comment (pmderodat) [reply src/e3/testsuite/report/rewriting.py:214]: Ah thanks! Rewriting summaries are actually a blind spot of the tests I’ve written. I’ll take care of fixing this and adding necessary tests, thanks!

**Pull request merged**